### PR TITLE
CFY-3691 - Make all internal REST calls secure

### PR DESCRIPTION
**Pull request by noakup**

### CODE REVIEW

File: plugins/riemann-controller/riemann_controller/resources/manager.config [hunk 1]
```diff
@@ -442,13 +442,18 @@
...
   (let [node-id               (:node-id ctx)
-        manager-ip            (or (System/getenv "MANAGEMENT_IP") "127.0.0.1")
-        raw-manager-rest-port (or (System/getenv "MANAGER_REST_PORT") "8101")
-        manager-rest-port     (Integer/parseInt raw-manager-rest-port)
-        base-uri              (str "http://" manager-ip ":" manager-rest-port "/api/v2")
+        rest-host     (or (System/getenv "REST_HOST") "127.0.0.1")
+        rest-protocol (or (System/getenv "REST_PROTOCOL") "http")
+        raw-rest-port (or (System/getenv "REST_PORT") "80")
+        rest-port     (Integer/parseInt raw-rest-port)
+        raw-verify-ssl-cert   (or (System/getenv "VERIFY_SSL_CERTIFICATE") "false")
```
Comment (dankilman) [inline plugins/riemann-controller/riemann_controller/resources/manager.config]: remove cert code

File: resources/rest-service/cloudify/triggers/execute_workflow.clj [hunk 2]
```diff
@@ -2,10 +2,13 @@
   [ctx]
     (let [deployment-id         (:deployment_id ctx)
           parameters            (:trigger-parameters ctx)
-          manager-ip            (or (System/getenv "MANAGEMENT_IP") "127.0.0.1")
-          raw-manager-rest-port (or (System/getenv "MANAGER_REST_PORT") "8101")
-          manager-rest-port     (Integer/parseInt raw-manager-rest-port)
-          base-uri              (str "http://" manager-ip ":" manager-rest-port "/api/v2")
+          rest-host     (or (System/getenv "REST_HOST") "127.0.0.1")
```
Comment (dankilman) [inline resources/rest-service/cloudify/triggers/execute_workflow.clj]: "

File: rest-service/manager_rest/server.py [hunk 3]
```diff
@@ -72,9 +71,9 @@ def setup_app(warnings=None):
         app.logger.info('initializing rest-service security')
         init_secured_app(app)
 
-    app.before_request(maintenance_mode_handler)
-
+    # TODO verify logging is executed before handling maintenance mode
```
Comment (dankilman) [inline rest-service/manager_rest/server.py]: remove todo

File: workflows/cloudify_system_workflows/snapshot.py [hunk 4]
```diff
@@ -571,8 +571,8 @@ def insert_agents_data(client, agents):
                 runtime_properties = node_instance.runtime_properties
                 old_agent = runtime_properties.get('cloudify_agent', {})
                 if not broker_config.get('broker_ip'):
-                    broker_config['broker_ip'] = old_agent.get('manager_ip',
-                                                               '')
+                    broker_config['broker_ip'] = \
+                        old_agent.get('manager_ip', '')
```
Comment (dankilman) [inline workflows/cloudify_system_workflows/snapshot.py]: verify agent migration still works as expected

File: rest-service/manager_rest/workflow_client.py [hunk 5]
```diff
@@ -16,13 +16,46 @@
...
+    def __init__(self):
+        cfy_config = config.instance()
+        cloudify_username = None
+        cloudify_password = None
+        verify_certificate = False
+        rest_port = DEFAULT_REST_PORT
+        rest_protocol = DEFAULT_REST_PROTOCOL
+        security_enabled = cfy_config.security_enabled
+        if security_enabled:
+            cloudify_username = cfy_config.security_admin_username
```
Comment (dankilman) [inline rest-service/manager_rest/workflow_client.py]: use agent user/password and update cfy_config

File: plugins/riemann-controller/riemann_controller/resources/manager.config [hunk 6]
```diff
@@ -442,13 +442,17 @@
...
+        raw-rest-port         (or (System/getenv "REST_PORT") "80")
+        rest-port             (Integer/parseInt raw-rest-port)
+        verify-ssl-cert       false
+        base-uri              (str rest-protocol "://" rest-host ":" rest-port "/api/v2")
         node-endpoint         (str "/node-instances/" node-id)
         node-resource-uri     (str base-uri node-endpoint)
-        get-node              (fn [] (clj-http.client/get node-resource-uri {:accept :json}))
+        get-node              (fn [] (clj-http.client/get node-resource-uri
+                                {:accept    :json
+                                :insecure?  (not= verify-ssl-cert true)}))
```
Comment (dankilman) [inline plugins/riemann-controller/riemann_controller/resources/manager.config]: can be simplified to `:insecure? (not verify-ssl-cert)`

File: resources/rest-service/cloudify/triggers/execute_workflow.clj [hunk 7]
```diff
@@ -19,4 +21,5 @@
          :accept         :json
          :socket-timeout (:socket_timeout parameters)
          :conn-timeout   (:conn_timeout parameters)
+         :insecure?      (not= verify-ssl-cert true)
```
Comment (dankilman) [inline resources/rest-service/cloudify/triggers/execute_workflow.clj]: can be simplified as shown previously

File: tests/mock_plugins/testmockoperations/tasks.py [hunk 8]
```diff
@@ -276,11 +276,11 @@ def get_instance_ip_of_source_and_target(ctx, **_):
         )
         data['mock_operation_invocation'].append((
             '{}_source'.format(ctx.source.node.name),
-            ctx.source.instance.host_ip
+            ctx.source.instance.private_ip
```
Comment (dankilman) [inline tests/mock_plugins/testmockoperations/tasks.py]: there is not such thing as instance.private_ip, restore to instance.host_ip

File: tests/mock_plugins/testmockoperations/tasks.py [hunk 9]
```diff
@@ -276,11 +276,11 @@ def get_instance_ip_of_source_and_target(ctx, **_):
...
         ))
         data['mock_operation_invocation'].append((
             '{}_target'.format(ctx.target.node.name),
-            ctx.target.instance.host_ip
+            ctx.target.instance.private_ip
```
Comment (dankilman) [inline tests/mock_plugins/testmockoperations/tasks.py]: there is not such thing as instance.private_ip, restore to instance.host_ip

File: tests/testenv/processes/manager_rest.py [hunk 10]
```diff
@@ -52,7 +53,8 @@ def __init__(self,
                  amqp_password,
                  maintenance_folder):
         self.process = None
-        self.port = port or MANAGER_REST_PORT
+        self.host = host or REST_HOST
```
Comment (dankilman) [inline tests/testenv/processes/manager_rest.py]: you are passing the host, and storing the port but they are never used.

Comment (noakup) [inline tests/testenv/processes/manager_rest.py]: port is used in `def start()` as well

File: rest-service/manager_rest/workflow_client.py [hunk 11]
```diff
@@ -21,8 +21,8 @@
 
 class WorkflowClient(object):
 
-    @staticmethod
-    def execute_workflow(name,
+    def execute_workflow(self,
```
Comment (dankilman) [inline rest-service/manager_rest/workflow_client.py]: restore

File: rest-service/manager_rest/workflow_client.py [hunk 12]
```diff
@@ -54,9 +54,8 @@ def execute_workflow(name,
                             execution_id=execution_id,
                             execution_parameters=execution_parameters)
 
-    @staticmethod
-    def execute_system_workflow(wf_id, task_id, task_mapping, deployment=None,
-                                execution_parameters=None,
+    def execute_system_workflow(self, wf_id, task_id, task_mapping,
```
Comment (dankilman) [inline rest-service/manager_rest/workflow_client.py]: restore

**Pull request merged**